Libplanet/Blockchain/BlockChain.cs

```diff
@@ -526,26 +526,34 @@ public void UnstageTransactions(ISet<Transaction<T>> transactions)
         /// <paramref name="address"/>.</returns>
         public long GetNextTxNonce(Address address)
         {
-            long nonce = Store.GetTxNonce(Id, address);
-            IEnumerable<Transaction<T>> stagedTxs = Store
-                .IterateStagedTransactionIds()
-                .Select(Store.GetTransaction<T>)
-                .Where(tx => tx.Signer.Equals(address))
-                .OrderBy(tx => tx.Nonce);
+            _rwlock.EnterReadLock();
 
-            long prevNonce = nonce - 1;
-            foreach (long n in stagedTxs.Select(tx => tx.Nonce))
+            try
             {
-                if (n != prevNonce + 1)
+                long nonce = Store.GetTxNonce(Id, address);
+                var stagedTxNonces = Store.IterateStagedTransactionIds()
+                    .Select(Store.GetTransaction<T>)
+                    .Where(tx => tx.Signer.Equals(address) && tx.Nonce >= nonce)
+                    .Select(tx => tx.Nonce)
+                    .OrderBy(n => n)
+                    .ToList();
```

Comment:
I guess it's okay that we can skip `.ToList()` because we use `stagedTxNonces` only once.